akka-actor/src/main/scala/akka/dispatch/Future.scala

```diff
@@ -80,13 +81,12 @@ object ExecutionContexts {
    * This is an execution context which runs everything on the calling thread.
    * It is very useful for actions which are known to be non-blocking and
    * non-throwing in order to save a round-trip to the thread pool.
+   *
+   * INTERNAL API
    */
-  private[akka] object sameThreadExecutionContext extends ExecutionContext with BatchingExecutor {
-    override protected def unbatchedExecute(runnable: Runnable): Unit = runnable.run()
-    override protected def resubmitOnBlock: Boolean = false // No point since we execute on same thread
-    override def reportFailure(t: Throwable): Unit =
-      throw new IllegalStateException("exception in sameThreadExecutionContext", t)
-  }
+  @InternalApi
+  private[akka] val sameThreadExecutionContext: ExecutionContext = akka.util.internal.SameThreadExecutionContext()
```

Comment:
I'm worried that this `sameThreadExecutionContext` is used outside of core Akka and that we will break binary compatibility because of this change. For example Alpakka is using it. We can deprecate the old and have a different name on this new.

Reply:
You mean because it is no longer publicly a `BatchingExecutor`? I'd expect that potential outside usage only are interested in it being an `ExecutionContext` (passing it to future as explicit executor) and that should be fine.

Reply:
source compatibility is probably fine, but not sure I dare to trust that the change is binary compatible

Reply:
I'll double check that using it as ec for future ops is bincomp. Can you think of any other possible usage (I can't)?

Reply:
Dang! You are right, the old object results in a class named `akka/dispatch/ExecutionContexts$sameThreadExecutionContext$` which is lost if I replace it with a val like this and for some reason that class is referenced where it is used (`map(n => n)(sameThreadExecutionContext)` for example) so it is not binary compatible. Deprecate+new name it is then.

Reply:
Introduced a new name `parasitic` (I don't like it but better to align with Scala 2.13 than making another new name up), replaced all internal usages  and deprecated the old.